404-server/server.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// A webserver that only serves a 404 page. Used as a default backend.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+)
+
+var port = flag.Int("port", 8080, "Port number to serve default backend 404 page.")
+
+func main() {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Fprint(w, "default backend - 404")
```

Comment:
Avenues for growth - add a defined volume mount point and let the user provide a template file and images.